tests/scenarios/k8s/base/mpi-job.yaml

```diff
@@ -0,0 +1,80 @@
+apiVersion: kubeflow.org/v2beta1
+kind: MPIJob
+metadata:
+  name: nccl-tests
+  labels:
+    app: nccl-tests
+spec:
+  slotsPerWorker: 8
+  runPolicy:
+    cleanPodPolicy: Running
+    activeDeadlineSeconds: 666
+  mpiReplicaSpecs:
+    Launcher:
+      replicas: 1
+      template:
+        metadata:
+          labels:
+            app: nccl-tests
+        spec:
+          restartPolicy: OnFailure
+          containers:
+          - image: ghcr.io/Azure/aks-rdma-infiniband/nccl-tests
+            name: nccl
+            env:
+            - name: OMPI_ALLOW_RUN_AS_ROOT
+              value: "1"
+            - name: OMPI_ALLOW_RUN_AS_ROOT_CONFIRM
+              value: "1"
+            command: ["/bin/bash", "-c"]
+            args:
+            - |
+              set -xe
+              until mpirun -np 16 -x LD_LIBRARY_PATH -bind-to none /usr/bin/nvidia-smi; do sleep 5; done
+              mpirun \
+                -np 16 \
```

Comment:
The number of process is dynamic based on how many GPUs each worker has claimed, since we have test cases for 1 and 8 GPUs

Reply:
Added a mechanism to let the job know the number of processes it should start.